custom-middleware/app-level/server.js

```diff
@@ -0,0 +1,69 @@
+import express from "express";
+import { StatusCodes } from "http-status-codes";
+
+const app = express();
+
+const authenticateUser = (req, res, next) => {
+  if (req.headers["x-username"]) {
+    req.username = req.headers["x-username"];
+  } else {
+    req.username = null;
+  }
+  next();
+};
+
+const passRequestPostBodyAsJSONArray = (req, res, next) => {
+  let body = "";
+  req.on("data", (chunk) => {
+    body += chunk;
+  });
+
+  req.on("end", () => {
+    try {
+      const jsonBody = JSON.parse(body);
+      if (
+        Array.isArray(jsonBody) &&
+        jsonBody.every((item) => typeof item === "string")
+      ) {
+        req.body = jsonBody;
+        next();
+      } else {
```

Comment:
praise: good error handling here, and handling different potential types in a graceful and readable way